Sources/PackageCollections/PackageIndexAndCollections.swift

```diff
@@ -234,43 +228,110 @@ public struct PackageIndexAndCollections: Closable {
     ///
     /// - Parameters:
     ///   - query: The search query
-    ///   - collections: Optional. If specified, only search within these collections.
+    ///   - in: Indicates whether to search in the index only, collections only, or both.
+    ///         The optional `Set<CollectionIdentifier>` in some enum cases restricts search within those collections only.
     ///   - callback: The closure to invoke when result becomes available
     public func findPackages(
         _ query: String,
-        collections: Set<PackageCollectionsModel.CollectionIdentifier>? = nil,
+        in searchIn: SearchIn = .both(collections: nil),
         callback: @escaping (Result<PackageCollectionsModel.PackageSearchResult, Error>) -> Void
     ) {
-        // Package index not available - fallback to collections
-        guard self.index.isEnabled else {
-            return self.collections.findPackages(query, collections: collections, callback: callback)
-        }
-        
-        self.index.findPackages(query) { indexResult in
```

Comment:
With the previous logic we can mimic index-only search by passing `collections: []`, but we can't mimic collections-only search since the result is built on top of what the index returns.